packages/clerk-auth/src/index.ts

```diff
@@ -0,0 +1,81 @@
+import type { ClerkOptions } from '@clerk/backend'
+import { Clerk, createIsomorphicRequest, constants } from '@clerk/backend'
+import type { Context, MiddlewareHandler } from 'hono'
+
+type ClerkAuth = Awaited<
+  ReturnType<ReturnType<typeof Clerk>['authenticateRequest']>
+>['toAuth'];
+
+declare module 'hono' {
+  interface ContextVariableMap {
+    clerk: ReturnType<typeof Clerk>;
+    clerkAuth: ReturnType<ClerkAuth>;
+  }
+}
+
+export const getAuth = (c: Context) => {
+  return c.get('clerkAuth')
+}
+
+export const clerkMiddleware = (options?: ClerkOptions): MiddlewareHandler => {
+  return async (c, next) => {
+    const { secretKey, publishableKey, apiUrl, apiVersion, ...rest } = options || {
+      secretKey: c.env.CLERK_SECRET_KEY || '',
```

Comment:
I realized that we need to use the `env` helper to support other runtimes, not only Cloudflare:

https://hono.dev/helpers/adapter#env

We can write it as follows:

```ts
    const {
      CLERK_PUBLISHABLE_KEY,
      CLERK_SECRET_KEY,
      CLERK_API_URL,
      CLERK_API_VERSION,
      CLERK_FRONTEND_API,
    } = env<Env>(c)
```